module/VuFind/tests/integration-tests/src/VuFindTest/Mink/LoggingTest.php

```diff
@@ -0,0 +1,267 @@
+                    'message_log'        => $this->getEmailLogPath(),
+                    'message_log_format' => $this->getEmailLogFormat(),
+                ],
+                'Logging' => [
+                    'email' => $emailConfig,
+                ],
+            ],
+        ]);
+
+        $this->resetEmailLog();
+
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl() . '/Search/Results?lookfor=test');
+        $page = $session->getPage();
+
+        // Wait for logging to complete
+        $this->findCss($page, 'body');
+
+        $loggedEmails = $this->getLoggedEmails();
+        $allEmailContent = implode('', array_map(fn ($email) => $email->toString(), $loggedEmails));
+        $allEmailSubjects = implode('', array_map(fn ($email) => $email->getSubject(), $loggedEmails));
+        $allEmailBodies = implode('', array_map(fn ($email) => $email->getBody()->getBody(), $loggedEmails));
+
+        // Basic assertions
+        $this->assertNotEmpty(
+            $allEmailContent,
+            $description . ': Expected to receive log email'
+        );
+
+        foreach ($expectedPatterns as $pattern) {
+            $this->assertMatchesRegularExpression(
+                $pattern,
+                $allEmailContent,
+                $description . ': Expected pattern not found: ' . $pattern
+            );
+        }
+
+        foreach ($unexpectedPatterns as $pattern) {
+            $this->assertDoesNotMatchRegularExpression(
+                $pattern,
+                $allEmailContent,
+                $description . ': Unexpected pattern found: ' . $pattern
+            );
+        }
+
+        // Email subject assertion
+        $this->assertStringContainsString(
+            'VuFind Log Message',
+            $allEmailSubjects,
+            'Email subject should contain "VuFind Log Message"'
+        );
+
+        // Conditional assertions based on log level/type
+        if (str_contains($emailConfig, 'debug')) {
+            $this->assertStringContainsString(
+                'DEBUG:',
+                $allEmailBodies,
+                'Email body should contain debug messages'
+            );
+
+            $this->assertStringContainsString(
+                'not-solr',
+                $allEmailBodies,
+                'Email body should contain the Solr URL that failed'
+            );
+        } else {
+            $this->assertStringContainsString(
+                'RequestErrorException',
+                $allEmailBodies,
+                'Email body should contain the specific exception type'
+            );
+        }
+
+        $this->assertStringContainsString(
+            '404 Not Found',
+            $allEmailBodies,
+            'Email body should contain the HTTP error'
+        );
+    }
+
+    /**
+     * Test that no emails are sent when logging is disabled
+     *
+     * @return void
+     */
+    public function testNoLoggingWhenDisabled(): void
```

Comment:
I'd suggest adjusting method names to reflect the "email" method, in case we want to add more test cases for other logging types in future:

```suggestion
    public function testNoEmailLoggingWhenDisabled(): void
```